tests/unit/s2n_async_pkey_test.c

```diff
@@ -63,6 +63,55 @@ static int async_handler_wipe_connection_and_apply(struct s2n_connection *conn)
     return S2N_FAILURE;
 }
 
+static int async_handler_sign_with_different_pkey_and_apply(struct s2n_connection *conn)
+{
```

Comment:
I'm still not seeing any solid proof that this handler (and therefore all your tests) runs. It should run, because it's the async handler, but unit tests are supposed to verify "should". 

If there's no existing state you could verify from main to prove these tests ran, you could set a flag in the callback and check that gets set.

Reply:
As per your previous recommendation, i updated the test case to 
`EXPECT_SUCCESS(try_handshake(server_conn, client_conn, async_handler_sign_with_different_pkey_and_apply));`

Try_handshake will ensure that all the steps are run. If that isn't the case then `try_handshake` call would fail after 5 retries and our expected value will differ. 

That should resolve above comment as well. 
Let me know your thoughts on it

Reply:
Done. Added `async_handler_called` ctr flag as a definitive proof for the test case